### PR TITLE
Friendly name update for IdentifiableNpmAuthorToken

### PR DESCRIPTION
**Pull request by HulonJenkins**
Renamed IdentifiableNpmAuthorToken to NpmAuthorIdentifiableToken to have consistency between Security and Security.Internal plugins.

### CODE REVIEW

File: Src/Plugins/Tests.Security/TestData/SecurePlaintextSecrets/ExpectedOutputs/SEC101_006.GitHubLegacyPat.sarif [hunk 1]
```diff
@@ -0,0 +1,280 @@
+{
```
Comment (marmegh) [inline Src/Plugins/Tests.Security/TestData/SecurePlaintextSecrets/ExpectedOutputs/SEC101_006.GitHubLegacyPat.sarif:1]: Is this sarif file addition expected with this change? Was it inadvertently excluded during a previous PR?

File: Src/ReleaseHistory.md [hunk 2]
```diff
@@ -8,6 +8,7 @@
 - Update `sarif-sdk` submodule to commit [7ddf923a4652a333f3356e9db4c5742b78a22c96](https://github.com/microsoft/sarif-sdk/commit/7ddf923a4652a333f3356e9db4c5742b78a22c96), which includes threading fixes in driver framework and fixes to ensure snipet population.
 - Update `spam` executable and dotnet library name to Sarif.PatternMatcher.Cli. 
 - Update `Microsoft.Security.Utilities` to [v1.4.0](https://github.com/microsoft/security-utilities/releases/tag/v1.4.0). [#662](https://github.com/microsoft/sarif-pattern-matcher/pull/662)
+- Rename IdentifiableNpmAuthorToken to NpmAuthorIdentifiableToken [#683](https://github.com/microsoft/sarif-pattern-matcher/pull/683)
```
Comment (marmegh) [inline Src/ReleaseHistory.md:11]: >IdentifiableNpmAuthorToken to NpmAuthorIdentifiableToken [](http://example.com/codeflow?start=9&length=56)

Almost there! Please include the RuleId (SEC101/050) and enclose the rule names in backticks.

Something like:
```
Rename `SEC101/050.IdentifiableNpmAuthorToken` to `SEC101/050.NpmAuthorIdentifiableToken` 
```

File: Src/Plugins/Security/SEC101.SecurePlaintextSecrets.json [hunk 3]
```diff
@@ -479,9 +479,10 @@
         },
         {
           "Id": "SEC101/050",
-          "Name": "DoNotExposePlaintextSecrets/IdentifiableNpmAuthorToken",
```
Comment (michaelcfanning) [inline Src/Plugins/Security/SEC101.SecurePlaintextSecrets.json:482]: >IdentifiableNpmAuthorToken [](http://example.com/codeflow?start=47&length=26)

The actual name of this rule is NpmAuthorToken, not sure how 'identifiable' was added. :)

**Review by michaelcfanning — APPROVED** [review]
:shipit:

**Pull request merged**